build_stage_repository

```diff
@@ -172,47 +168,167 @@ def packages_from_comps(comps):
     return packages
 
 
-def main():
-    try:
-        collection = sys.argv[1]
-        version = sys.argv[2]
-        dist = sys.argv[3]
-        arch = 'x86_64'
-    except IndexError:
-        raise SystemExit(f"Usage: {sys.argv[0]} collection version os")
+def copr_repository(collection, version, dist, arch):
+    return f"https://download.copr.fedorainfracloud.org/results/@theforeman/{collection}-{version}-staging/rhel-{dist.replace('el', '')}-{arch}"
 
-    base_dir = 'tmp'
-    rpm_sync_dir = f"{base_dir}/rpms"
-    srpm_sync_dir = f"{base_dir}/srpms"
 
-    stage_dir = f"{base_dir}/{collection}/{version}/{dist}/"
-    rpm_dir = f"{stage_dir}/{arch}"
-    srpm_dir = f"{stage_dir}/source"
+def prod_repository(collection, version, dist, arch):
+    if collection == "foreman":
+        subdir = "releases"
+    else:
+        subdir = collection.replace('foreman-', '')
 
-    if not os.path.exists(rpm_sync_dir):
-        os.makedirs(rpm_sync_dir)
+    return f"https://yum.theforeman.org/{subdir}/{version}/{dist}/{arch}"
 
-    if not os.path.exists(srpm_sync_dir):
-        os.makedirs(srpm_sync_dir)
 
-    if not os.path.exists(rpm_dir):
-        os.makedirs(rpm_dir)
+def copr_repository_urls(repository):
+    cmd = [
+        'dnf',
+        'reposync',
+        '--urls',
+        '--refresh',
+        '--repofrompath',
+        f"copr,{repository}",
+        '--repo',
+        'copr'
+    ]
 
-    if not os.path.exists(srpm_dir):
-        os.makedirs(srpm_dir)
+    urls = check_output(cmd, universal_newlines=True, stderr=STDOUT)
+    return urls.split("\n")
+
+
+def compare_repositories(new_repository, old_repository, source=False):
+    cmd = [
+        'dnf',
+        'repodiff',
+        '--simple',
+        '--refresh',
+        '--compare-arch',
+        '--repofrompath',
+        f"new,{new_repository}",
+        '--repofrompath',
+        f"old,{old_repository}",
+        '--repo-old',
+        'old',
+        '--repo-new',
+        'new'
+    ]
+
+    if not source:
+        cmd.extend(['--arch', 'noarch,x86_64'])
+
+    print(' '.join(cmd))
+    return check_output(cmd, universal_newlines=True)
+
+
+def parse_repodiff(diff):
+    split_diff = diff.split("\n")
+    packages = []
+
+    for line in split_diff:
+        if line.startswith('Added package  :'):
+            packages.append(line.replace('Added package  : ', ''))
+
+        if ' -> ' in line:
+            packages.append(line.split(' -> ')[1])
+
+    return packages
+
+
+def download_copr_packages(packages, urls, repository, downloads_dir, included_packages):
+    for package in packages:
+        include = any(included in package for included in included_packages)
+
+        if include:
+            name = f"{package}.rpm".replace("-1:", "-")
+
+            for url in urls:
+                if name in url:
```

Comment:
I'm wondering about strict matching. Should we at least make it
```suggestion
                if url.endswith(name):
```

I also considered exact matching:
```suggestion
                if name == os.path.basename(url):
```